api/src/pcapi/core/educational/models.py

```diff
@@ -794,46 +813,65 @@ def displayedStatus(self) -> CollectiveOfferDisplayedStatus:
-                return CollectiveOfferDisplayedStatus.PREBOOKED
-            if last_booking_status == CollectiveBookingStatus.CANCELLED:
-                return CollectiveOfferDisplayedStatus.CANCELLED
-
+        match self.validation:
+            case offer_mixin.OfferValidationStatus.DRAFT:
+                return CollectiveOfferDisplayedStatus.DRAFT
+            case offer_mixin.OfferValidationStatus.PENDING:
+                return CollectiveOfferDisplayedStatus.PENDING
+            case offer_mixin.OfferValidationStatus.REJECTED:
+                return CollectiveOfferDisplayedStatus.REJECTED
+            case offer_mixin.OfferValidationStatus.APPROVED:
+                last_booking_status = self.lastBookingStatus
+                has_booking_limit_passed = self.hasBookingLimitDatetimesPassed
+                has_started = self.hasBeginningDatetimePassed
+                has_ended = self.hasEndDatetimePassed
+
+                match last_booking_status:
+                    case None:
+                        if has_booking_limit_passed:
+                            if has_started and feature.FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
+                                return CollectiveOfferDisplayedStatus.CANCELLED
+                            return CollectiveOfferDisplayedStatus.EXPIRED
+
+                        return CollectiveOfferDisplayedStatus.ACTIVE
+
+                    case CollectiveBookingStatus.PENDING:
+                        # pylint: disable=using-constant-test
+                        if has_booking_limit_passed:
+                            return CollectiveOfferDisplayedStatus.EXPIRED
+                        return CollectiveOfferDisplayedStatus.PREBOOKED
+
+                    case CollectiveBookingStatus.CONFIRMED:
+                        # pylint: disable=using-constant-test
+                        if has_ended:
+                            return CollectiveOfferDisplayedStatus.ENDED
+                        return CollectiveOfferDisplayedStatus.BOOKED
+
+                    case CollectiveBookingStatus.USED:
+                        return CollectiveOfferDisplayedStatus.ENDED
+
+                    case CollectiveBookingStatus.REIMBURSED:
+                        if feature.FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
+                            return CollectiveOfferDisplayedStatus.REIMBURSED
+                        return CollectiveOfferDisplayedStatus.ENDED
+
+                    case CollectiveBookingStatus.CANCELLED:
+                        if feature.FeatureToggle.ENABLE_COLLECTIVE_NEW_STATUSES.is_active():
+                            if (
+                                self.lastBookingCancellationReason == CollectiveBookingCancellationReasons.EXPIRED
+                                and not has_started
+                            ):
+                                # There is script that set the booking status to EXPIRED when the booking is expired.
```

Comment:
`... booking status to CANCELLED ...`

(c'est la cancellationReason qui est EXPIRED)